### PR TITLE
Add types to constructors and private/final/internal methods (Batch I)

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32179, #33228
| License       | MIT
| Doc PR        | N/A

As promised, now a larger batch with the following components:
* Asset
* BrowserKit
* Config
* Console
* ~~CssSelector~~
* Debug
* ~~DomCrawler~~
* DotEnv
* ErrorHandler
* ErrorRenderer
* ExpressionLanguage
* ~~Filesystem~~
* ~~Finder~~

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Do we really want to add types to `Tests`? I'm anticipating hours on merge conflicts here. On non-test classes, it's worth it, but on tests, I'd prefer doing anything else of my future contrib time...

**Comment by stof** [comment]
I agree with @nicolas-grekas here.

**Comment by derrabus** [comment]
> Do we really want to add types to `Tests`?

I find them helpful for understanding the tests.

> I'm anticipating hours on merge conflicts here. On non-test classes, it's worth it, but on tests, I'd prefer doing anything else of my future contrib time...

I understand. Suggestion: I‘m going to remove changes I‘ve made to test case classes that also exist on the 3.4 branch. We can hold them back until 3.4 is EOL.

**Comment by nicolas-grekas** [comment]
>  I‘m going to remove changes I‘ve made to test case classes that also exist on the 3.4 branch. We can hold them back until 3.4 is EOL.

that'd work for me (but if it's too tedious, you can surrender :)

**Comment by nicolas-grekas** [comment]
Status: needs work

**Comment by derrabus** [comment]
Changes to tests have been moved into a separate branch: https://github.com/derrabus/symfony/tree/improvement/typed-tests

I've kept the changes to the Error* components though because those components are new and thus shouldn't cause merge conflicts.

**Comment by derrabus** [comment]
Status: Needs Review

File: src/Symfony/Component/Console/Helper/Table.php [hunk 1]
```diff
@@ -580,7 +580,7 @@ private function buildTableRows(array $rows)
             }
         }
 
-        return new TableRows(function () use ($rows, $unmergedRows) {
+        return new TableRows(function () use ($rows, $unmergedRows): \Traversable {
```
Comment (xabbuh) [inline src/Symfony/Component/Console/Helper/Table.php]: `iterable` like in the other places?

Comment (derrabus) [reply src/Symfony/Component/Console/Helper/Table.php]: The `TableRows` class requires a callable that returns a real `\Traversable`. This is why I did not use `iterable` as return type here.

File: src/Symfony/Component/Console/Application.php [hunk 2]
```diff
@@ -1125,12 +1125,12 @@ public function setDefaultCommand($commandName, $isSingleCommand = false)
...
      * @internal
      */
-    public function isSingleCommand()
+    public function isSingleCommand(): bool
     {
         return $this->singleCommand;
     }
 
-    private function splitStringByWidth(string $string, int $width)
+    private function splitStringByWidth(string $string, int $width): array
```
Comment (xabbuh) [inline src/Symfony/Component/Console/Application.php]: Theoretically, `str_split()` below can return `false` (looking at callers of the `splitStringByWidth()` that could happen when the terminal width is `5` or less. How do we deal with that?

Comment (derrabus) [reply src/Symfony/Component/Console/Application.php]: This is the only call that I've found. If `splitStringByWidth` really returned `false` we would run into an error anyway.

https://github.com/symfony/symfony/blob/a0bbae75140584e51237c39b90ced54fdd261016/src/Symfony/Component/Console/Application.php#L805

Comment (xabbuh) [reply src/Symfony/Component/Console/Application.php]: Yes, that's the call I referred to. But you are right.That's already broken. So only the error message changes. We may still need to think about fixing it. That's out of the scope of this PR though.

**Review by xabbuh — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @derrabus.

**Pull request merged**